src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1938,6 +2101,165 @@ storePointsToValue opts cc env tyenv nameEnv base_mem maybe_cond ptr val maybe_i
+  Map AllocIndex (MS.AllocSpec (LLVM arch)) ->
+  Map AllocIndex (MS.TypeName (LLVM arch)) ->
+  Crucible.MemImpl Sym ->
+  LLVMPtr (Crucible.ArchWidth arch) ->
+  BitfieldIndex ->
+  SetupValue (LLVM arch) ->
+  IO (Crucible.MemImpl Sym)
+storePointsToBitfieldValue opts cc env tyenv nameEnv base_mem ptr bfIndex val = do
+  let sym = cc ^. ccBackend
+
+  let alignment = Crucible.noAlignment -- default to byte alignment (FIXME)
+
+  smt_array_memory_model_enabled <- W4.getOpt
+    =<< W4.getOptionSetting enableSMTArrayMemoryModel (W4.getConfiguration sym)
+
+  -- Unlike in storePointsToValue, we compute the MemTy/StorageType not from
+  -- the RHS value, but from the BitfieldIndex. This is because we need to
+  -- load the entire bitfield, which can be larger than the size of the RHS
+  -- value.
+  let memTy = biBitfieldType bfIndex
+  storTy <- Crucible.toStorableType memTy
+
+  case val of
+    SetupTerm _tm
+      -- TODO RGS: File an issue about this
+      | smt_array_memory_model_enabled ->
+        fail $ unlines
+          [ "llvm_points_to_bitfield currently does not work in"
+          , "combination with the enable_smt_array_memory_model option."
+          ]
+    _ -> do
+      -- Resolve the RHS value as an LLVMVal.
+      rhsVal <- X.handle (handleException opts) $
+        resolveSetupVal cc base_mem env tyenv nameEnv val
+
+      -- Load the bitfield that `ptr` points to.
+      bfLoadedVal <- Crucible.loadRaw sym base_mem ptr storTy alignment
+
+      -- Assert that the bitfield was loaded successfully. If not, abort.
+      let badLoadSummary = summarizeBadLoad cc memTy PreState ptr
+      bfVal <- case bfLoadedVal of
+        Crucible.NoErr p res_val -> do
+          let rsn = Crucible.AssertFailureSimError "Error loading bitvector" $
+                    show badLoadSummary
+          Crucible.assert sym p rsn
+          pure res_val
+        Crucible.Err _p ->
+          fail $ show $ describeConcreteMemoryLoadFailure base_mem badLoadSummary ptr
+
+      case (bfVal, rhsVal) of
+        -- This will only work if:
+        --
+        -- * Both the bitfield and the RHS value are bitvectors, and
+        -- * The width of the RHS type is less than or equal to the width
+        --   of the bitfield type.
+        --
+        -- We check these criteria in this case.
+        (Crucible.LLVMValInt bfBlk bfBV, Crucible.LLVMValInt _rhsBlk rhsBV)
+          -> let bfWidth = W4.bvWidth bfBV
+                 rhsWidth = W4.bvWidth rhsBV in
+             case testLeq (incNat rhsWidth) bfWidth of
+               Nothing ->
+                 fail $ unlines
+                   [ "llvm_points_to_bitfield: RHS value's size must be less then or equal to bitfield's size"
+                   , "Bitvector width: " ++ show bfWidth
+                   , "RHS value width: " ++ show rhsWidth
+                   ]
+               Just LeqProof ->
+                 do -- Here is where we perform the bit-twiddling needed
```

Comment:
All of this bit-twiddling could be made more efficient if we grouped up `llvm_points_to_bitfield` commands on the same bitfield and collapsed them into a single bitmask. This PR does not implement this idea, however. I should open an issue to track doing so.

Reply:
I've opened #1541 and referenced it from the comments at the bottom of this function.